flink-libraries/flink-state-processing-api/src/test/java/org/apache/flink/state/table/module/ExampleDynamicBuiltInFunctionDefinitionFactory.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.state.table.module;
+
+import org.apache.flink.table.functions.BuiltInFunctionDefinition;
+import org.apache.flink.table.functions.DynamicBuiltInFunctionDefinitionFactory;
+
+import java.util.List;
+
+public class ExampleDynamicBuiltInFunctionDefinitionFactory
```

Comment:
api looks minimal and clear.
nit: maybe we can standardize to <module>.<provider> and document that function names returned must be unique within the module (and what happens otherwise)
nit: maybe we can add an optional `ModuleContext` param (classloader, config). not needed now, but helps if a provider later needs context without breaking the spi

Reply:
makes sense, thanks for clarifying
on the “built-in function name suggestion”: understood, it’s internal and long-standing, so I’m fine relying on it. if feasible, a short internal comment pointing to where that suggestion lives would help new contributors

bottom line: +1 to keep the api minimal as proposed